Framework/Algorithms/src/Integration.cpp

```diff
@@ -242,10 +246,20 @@ void Integration::exec() {
 
     double sumY = 0.0;
     double sumE = 0.0;
+    double sumF = 0.0;
+    double Fmin = 0.0;
+    double Fmax = 0.0;
     if (distmax <= distmin) {
       sumY = 0.;
       sumE = 0.;
     } else {
+      if (rebinned_input) {
+        // Workspace has fractional area information, need to take into account
+        const MantidVec &F = rebinned_input->readF(i);
+        sumF = std::accumulate(F.begin() + distmin, F.begin() + distmax, 0.0);
+        Fmin = F[distmin - 1];
```

Comment:
If no limits are set I think `distmin=0` doesn't it? This line would then be invalid

Reply:
Yes. I'll add a check for `distmin > 0`